File: openapi/code_samples/python/api_locations_{postalCode}_getCityDetailsByPostalCode.py
from flask import Flask, jsonify
app = Flask(__name__)

@app.route('/api/location/findByPostalCode/<postalCode>', methods=['GET'])
def find_by_postal_code(postalCode):
    # Your implementation here to retrieve city details by postal code
    # Simulated data for demonstration purposes
    cities = [
        {"Name": "City 1", "PostalCode": postalCode},
        {"Name": "City 2", "PostalCode": postalCode},
        {"Name": "City 3", "PostalCode": postalCode}
    ]

    return jsonify(cities)

if __name__ == '__main__':
    app.run()

File: openapi/code_samples/javascript/api_locations_{postalCode}_getCityDetailsByPostalCode.js
const express = require('express');
const app = express();
const port = 3000;

app.get('/api/location/findByPostalCode/:postalCode', (req, res) => {
    const postalCode = req.params.postalCode;

    // Your implementation here to retrieve city details by postal code
    // Simulated data for demonstration purposes
    const cities = [
        { Name: "City 1", PostalCode: postalCode },
        { Name: "City 2", PostalCode: postalCode },
        { Name: "City 3", PostalCode: postalCode }
    ];

    res.status(200).json(cities);
});

app.listen(port, () => {
    console.log(`Server is listening at http://localhost:${port}`);
});
